src/loaders/anndata-loaders/MatrixZarrLoader.js

```diff
@@ -0,0 +1,180 @@
+   * @returns {Object} A { data: Float32Array } contianing the CellXGene matrix.
+   */
+  async _loadCSRSparseCellXGene(matrix, shape) {
+    const { store } = this;
+    const { data: rows } = await openArray({ store, path: `${matrix}/indptr`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const { data: cols } = await openArray({ store, path: `${matrix}/indices`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const { data: cellXGene } = await openArray({ store, path: `${matrix}/data`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const cellXGeneMatrix = new Float32Array(shape[0] * shape[1]).fill(0);
+    let row = 0;
+    rows.forEach((_, index) => {
+      const rowStart = rows[index];
+      const rowEnd = rows[index + 1];
+      for (let i = rowStart; i < rowEnd; i += 1) {
+        const val = cellXGene[i];
+        const col = cols[i];
+        cellXGeneMatrix[row * shape[1] + col] = val;
+      }
+      row += 1;
+    });
+    return { data: cellXGeneMatrix };
+  }
+
+  /**
+   * Class method for loading column oriented (CSC) sparse data from zarr.
+   * @param {string} matrix Location of the matrix.
+   * @param {string} shape Shape of the non-sparse output.
+   * @returns {Object} A { data: Float32Array } contianing the CellXGene matrix.
+   */
+  async _loadCSCSparseCellXGene(matrix, shape) {
+    const { store } = this;
+    const { data: cols } = await openArray({ store, path: `${matrix}/indptr`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const { data: rows } = await openArray({ store, path: `${matrix}/indices`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const { data: cellXGene } = await openArray({ store, path: `${matrix}/data`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const cellXGeneMatrix = new Float32Array(shape[0] * shape[1]).fill(0);
+    let col = 0;
+    cols.forEach((_, index) => {
+      const colStart = cols[index];
+      const colEnd = cols[index + 1];
+      for (let i = colStart; i < colEnd; i += 1) {
+        const val = cellXGene[i];
+        const row = rows[i];
+        cellXGeneMatrix[row * shape[1] + col] = val;
+      }
+      col += 1;
+    });
+    return { data: cellXGeneMatrix };
+  }
+
+  /**
+   * Class method for loading the cell x gene matrix.
+   * @returns {Promise} A promise for the zarr array contianing the cell x gene data.
+   */
+  async loadCellXGene() {
+    const { store, url } = this;
+    if (this.arr) {
+      return this.arr;
+    }
+    const {
+      options: { matrix },
+    } = this;
+    const res = await fetch(`${url}/${matrix.replace('.', '/')}/.zattrs`);
```

Comment:
same here with `getJson`. The `store` handles checking for a `404` and will throw a `KeyError` if the item is missing:

```javascript
try { 
  const zattrs = getJson(this.store, `${matrix.replace('.', '/')}/.zattrs`);
  /* ... */
} catch (err) {
  if (!(err instanceof KeyError) {
     throw err; // rethrow error if not KeyError
  }
  // otherwise res was a 404 and don't do anything..
}
```